smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java

```diff
@@ -65,11 +65,17 @@ public KafkaSource(Vertx vertx, Config config, String servers) {
                 .doOnError(t -> LOGGER.error("Unable to read a record from Kafka topic '{}'", topic, t));
 
         if (config.getOptionalValue("retry", Boolean.class).orElse(true)) {
-            Integer max = config.getOptionalValue("retry-attempts", Integer.class).orElse(5);
-            flowable = flowable
+            Integer max = config.getOptionalValue("retry-attempts", Integer.class).orElse(null);
```

Comment:
Minor - I generally use -1 for infinite, but `null` is ok.

Reply:
Noted! Will change it to -1